### PR TITLE
Remove EFB to RAM cache, and simplify code.

### PR DESCRIPTION
**Pull request by magumagu**
degasus killed it; this is just the funeral.  (I would suggest appending "?w=1" to the GitHub URL if you want to review this.)

### CODE REVIEW

**Comment by degasus** [comment]
Through I'm fine with removing the efb2ram cache, I don't know when I've killed this feature.

btw: Half of your code does remove the killed (by myself) optimization to not upscale efb2tex when it was changed in the last time. Did I kill both?

File: Source/Core/VideoCommon/TextureCacheBase.h [hunk 1]
```diff
@@ -104,7 +103,7 @@ class TextureCache
 
 		bool OverlapsMemoryRange(u32 range_address, u32 range_size) const;
 
-		bool IsEfbCopy() { return (type == TCET_EC_VRAM || type == TCET_EC_DYNAMIC); }
+		bool IsEfbCopy() { return type == TCET_EC_VRAM; }
```
Comment (degasus) [inline Source/Core/VideoCommon/TextureCacheBase.h]: You can also replace this function with config.rendertarget

File: Source/Core/VideoCommon/VideoConfig.cpp [hunk 2]
```diff
@@ -199,8 +198,6 @@ void VideoConfig::GameIniLoad()
 	CHECK_SETTING("Video_Hacks", "EFBCopyEnable", bEFBCopyEnable);
 	CHECK_SETTING("Video_Hacks", "EFBToTextureEnable", bCopyEFBToTexture);
 	CHECK_SETTING("Video_Hacks", "EFBScaledCopy", bCopyEFBScaled);
-	CHECK_SETTING("Video_Hacks", "EFBCopyCacheEnable", bEFBCopyCacheEnable);
-	CHECK_SETTING("Video_Hacks", "EFBEmulateFormatChanges", bEFBEmulateFormatChanges);
```
Comment (degasus) [inline Source/Core/VideoCommon/VideoConfig.cpp]: Why also format changes?

**Comment by mimimi085181** [comment]
@magumagu: Can you give an example where the efb2ram cache is broken? I have been trying to fix bugs in it, and i thought it works better now than it did for a long time. One small issue i could imagine would be when using OpenGL and a texture uses a width or height that can't be divided by 4.

If this is removed, can i add a hack that breaks 50% of the efb2ram games, but makes the other 50% faster? And use the same spot in the GUI?

File: Source/Core/VideoCommon/VideoConfig.cpp (hunk 2 above)
Comment (magumagu) [inline Source/Core/VideoCommon/VideoConfig.cpp]: Oops, I didn't mean to do that.

**Comment by magumagu** [comment]
@degasus You killed it in 9f13a77: the cache never hits because we free the texture before we search for it.

@mimimi085181 It's "broken" in the sense that it doesn't do anything, not in the sense that it breaks any games.  If you want to add a different hack, feel free to propose it... I mean, UI space isn't exactly a precious commodity.

**Comment by magumagu** [comment]
Updated to address review comments.

**Comment by mimimi085181** [comment]
Does this actually work:
bool IsEfbCopy() { return config.rendertarget; }

I don't understand where it gets its info from, but i'm assuming it only works on the current texture/efb copy that is being processed. In OverlapsMemoryRange, IsEfbCopy() is called for random texture cache entries, does it still work properly there?

**Comment by degasus** [comment]
The cache was to not invalidate other textures in the efb2ram call. The name was just very confusing. So I still doubt I've killed the cache. Through I'm fine with removing it.

LGTM

**Comment by degasus** [comment]
@mimimi085181 The "config" member is defined on creating the texture. It's const and used for the pool. So it's always defined for every entry. Through it only stores the backend specific information and so it shouldn't be used for emulation. But atm we only use rendertarget textures for efb copys.

**Comment by Linktothepast** [comment]
It seems to cause this d3d only issue in Tales of Symphonia world map: http://prntscr.com/5xyime

**Pull request merged**